sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathClientBuilder.java

```diff
@@ -129,8 +139,12 @@ public DataLakePathClientBuilder() {
      * and {@link #retryOptions(RequestRetryOptions)} have been set.
      */
     public DataLakeFileClient buildFileClient() {
-        return new DataLakeFileClient(buildFileAsyncClient(),
-            blobClientBuilder.buildClient().getBlockBlobClient());
+        String dataLakeFileSystemName = CoreUtils.isNullOrEmpty(fileSystemName)
+            ? DataLakeFileSystemClient.ROOT_FILESYSTEM_NAME : fileSystemName;
+
+        return new DataLakeFileClient(buildFileAsyncClient(), blobClientBuilder.buildClient().getBlockBlobClient(),
```

Comment:
Same comment as the service client builder where this can end up creating multiple HttpPipelines.